### ISSUE TITLE
getObject returning a body of empty data

### ISSUE THREAD

**Issue by wrporter** [aws/aws-sdk-java#1392]
SDK Version 1.11.115
JDK 8

A very small amount (~0.002%) of our `getObject` requests are consistently returning bodies with empty data. 

Also interesting to note, we are uploading files and specifying a content type. Whenever the empty body is returned, the content type does not come back properly and is `application/octet-stream` rather than the original `application/pdf`, etc.

When I check the files manually through the AWS Console, the files download just fine.

Why would this be so intermittent? Is there a bug with S3?

**Comment by dagnir** [comment]
@wrporter I assume you're reading the entire object stream returned by S3? Do you receive either an MD5 validation error or length mismatch error?

**Comment by wrporter** [comment]
@dagnir Yes, we're reading the entire object stream. We don't receive any errors.

The code looks much like the following (simplified).

```java
AmazonS3 client = s3ClientRepository.getClientForBucket(bucket);
GetObjectRequest amazonRequest = new GetObjectRequest(bucket, key);
S3Object s3Attachment = client.getObject(amazonRequest);
byte[] data = com.google.common.io.ByteStreams.toByteArray(s3Attachment.getObjectContent());
```

Again, this is only happening for about 0.002% of our requests, out of hundreds of thousands of requests in a 30-day period. It is a small amount, but very consistent, occurring daily.

**Comment by dagnir** [comment]
Can you log what [`ObjectMetadata#getContentLength()`](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/ObjectMetadata.html#getContentLength--)? From your description it sounds like the problem is originating from S3's side since neither the MD5 or length checks are failing.

**Comment by wrporter** [comment]
I will make some changes to log some extra data and get back to you on that.

**Comment by dagnir** [comment]
Thanks!

**Comment by wrporter** [comment]
@dagnir Sorry for the delay. I added some logging. `ObjectMetadata#getContentLength()` returns `0` and `ObjectMetadata#getContentMD5()` returns `null`.

**Comment by dagnir** [comment]
@wrporter Thank you. It looks like the problem is on the service side S3 is sending you 0-length objects when you're not expecting them to be. Can you open a [support case ticket](https://console.aws.amazon.com/support/) in the AWS Console?

At this point I'm going to go ahead and close this issue since it doesn't look like the problem is being caused by the SDK. Please feel free to reopen if you require further assistance with this.

**Comment by Bwvolleyball** [comment]
I realize this is a _really_ old support ticket, but, was there a change to the aws-sdk-java published for this since @dagnir mentioned that

> It looks like the problem is on the service side S3 is sending you 0-length objects when you're not expecting them to be.

We're using a version of the aws-sdk-java newer than referenced in this ticket, but one that was still published before this ticket was opened (v1.1.200 from Sept 2017).

Our particular issue seems to be related to uploading an object and then immediately downloading it again, I'm not sure if OP's situation was similar or not, but under these circumstances our content length of the retrieved file is 0.

**Comment by narendranjs** [comment]
I am using the version 1.11.860

with the following code 
> S3Object object = s3Config.amazonS3Client().getObject(obj);
> S3ObjectInputStream stream = object.getObjectContent();
> byte[] content = IOUtils.toByteArray(stream);


For me the byte array is getting as null.
Tried to download the file after few mins and after few hours still the byte array is empty

Any suggestions are welcome
